### PR TITLE
72 happy path first

### LINKED ISSUE #72: Rule violation: Happy path first
### Describe the Bug

Need to resolve violation of the "happy path first" rule:

image
image

### Steps to Reproduce

1. model the diagram

### Expected Behavior
non-happy path connection is below happy path. 

### Environment

- Host (Browser/Node version), if applicable: [e.g. MS Edge 18, Chrome 69, Node 10 LTS]
- OS: [e.g. Windows 7]
- Library version: [e.g. 2.0.0]

### PR DESCRIPTION
**Pull request by abdul99ahad**
Closes #72 
### Proposed Changes

 
The primary path will always come first, with secondary connections following below.
image

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [x] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

### CODE REVIEW

File: lib/utils/layoutUtil.js [hunk 1]
```diff
@@ -120,6 +120,15 @@ export function connectElements(source, target, layoutGrid) {
...
 
+  // negative dX indicates connection from future to past
+  if (dX < 0 && dY <= 0) {
+    return [
+      getDockingPoint(sourceMid, sourceBounds, 'b'),
+      { x: sourceMid.x, y: sourceMid.y + DEFAULT_CELL_HEIGHT / 2 },
+      { x: targetMid.x, y: sourceMid.y + DEFAULT_CELL_HEIGHT / 2 },
+      getDockingPoint(targetMid, targetBounds, 'b')
+    ];
+  }
```
Comment (abdul99ahad) [inline lib/utils/layoutUtil.js:131]: @nikku Not exactly sure if this is the right approach. I have added the condition when it's a loop (future to past) means dX will be negative and the connection can be either on the same row or row less than the current element (dY <=0). 

For reference, It solves this case.
image

**Comment by abdul99ahad** [comment]
@nikku Confused about the strategy of happy path first after Till's comment https://github.com/bpmn-io/bpmn-auto-layout/issues/50#issuecomment-2377248858. Do we consider connections from top or not?

**Comment by nikku** [comment]
> Do we consider connections from top or not?

I consider that we fully understand our strategy, and take it from there. For me "back loop on top" is nothing that I've seen applied in practice, and it makes it harder to understand the happy path.

**Review by nikku — APPROVED** [review]
@abdul99ahad I separated spec and snapshot updates on the PR. This way it was easier to me to review before/after state.

This is a simple, yet effective, change to lead to overall cleaner diagrams, cf. diffs:

image

image

image

Hence, let's move forward with it, and have additional discussions as a follow-up.

**Pull request merged**